org.eclipse.jgit/src/org/eclipse/jgit/util/PGPSign.java

```diff
@@ -0,0 +1,160 @@
+package org.eclipse.jgit.util;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Iterator;
+
+import org.bouncycastle.bcpg.ArmoredOutputStream;
+import org.bouncycastle.bcpg.BCPGOutputStream;
+import org.bouncycastle.bcpg.HashAlgorithmTags;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openpgp.PGPException;
+import org.bouncycastle.openpgp.PGPUtil;
+import org.bouncycastle.openpgp.PGPPrivateKey;
+import org.bouncycastle.openpgp.PGPSecretKey;
+import org.bouncycastle.openpgp.PGPSecretKeyRing;
+import org.bouncycastle.openpgp.PGPSecretKeyRingCollection;
+import org.bouncycastle.openpgp.PGPSignature;
+import org.bouncycastle.openpgp.PGPSignatureGenerator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaKeyFingerprintCalculator;
+import org.bouncycastle.openpgp.operator.jcajce.JcaPGPContentSignerBuilder;
+import org.bouncycastle.openpgp.operator.jcajce.JcePBESecretKeyDecryptorBuilder;
+import org.eclipse.jgit.annotations.NonNull;
+
+/**
+ * @author mprabhala
+ * @since 5.2
+ *
+ */
+public class PGPSign {
```

Comment:
It's usually preferred to make a class name a noun. This is a PGPSigner, right?

Reply:
@bmp4070 let's leave it for now - however, this probably needs a better name and api / abstraction (similar to `SshSessionFactory` but less fancy). EGit very likely wan't to override it with something reading configuration from Eclipse preferences. However, JGit ususally just works with an out of box configuration using the user's envrionment.

But we'll address this in a later commit.